src/model/schema/ThunderstoreSchema.ts

```diff
@@ -0,0 +1,131 @@
+import Ajv from "ajv";
+import addFormats from "ajv-formats";
+import schema from "../../../schema.json";
+import schemaSchema from "../../../schemaSchema.json";
+import { GameSelectionDisplayMode } from "../game/GameSelectionDisplayMode";
+import R2Error from "../errors/R2Error";
+
+// Disabled for now, will become relevant when we migrate away from JSON modules.
+export interface Schema {
+    schema_version: string,
+    games: [string:SchemaEntry],
+    communities: [string:ThunderstoreEntry],
+    packageInstallers: [],
+    modloaderPackages: [ModloaderPackage],
+}
+
+export interface SchemaEntry {
+    uuid: string,
+    label: string,
+    meta: EntryMeta,
+    distributions: [EntryDist],
+    thunderstore?: ThunderstoreEntry,
+    r2modman?: EntryR2,
+}
+
+export interface ThunderstoreEntry {
+    displayName: string,
+    discordUrl?: string,
+    wikiUrl?: string,
+    categories: any,
+    sections: any,
+}
+
+export interface ModloaderPackage {
+    identifier: string,
+    rootFolder: string,
+    variant: string,
+}
+
+export interface EntryMeta {
+    displayName: string,
+    iconUrl: string,
+}
+
+export interface EntryDist {
+    platform: DistributionPlatform,
+    identifier?: string,
+}
+
+export interface EntryR2 {
+    internalFolderName: string,
+    dataFolderName?: string,
+    settingsIdentifier: string,
+    packageIndex: string,
+    exclusionsUrl: string,
+    steamFolderName: string,
+    exeNames: [string],
+    gameInstanceType: GameInstanceType,
+    gameSelectionDisplayMode: GameSelectionDisplayMode,
```

Comment:
Same as above, except the way `GameSelectionDisplayMode` is defined, the acceptable values are actually 0 and 1 while the data contains "visible" and "hidden".